In [1]:
# Import libraries

import pandas as pd
import numpy as np
import os

In [3]:
# 3. Import data
df_ords_prods_combined = pd.read_pickle(r'/Users/jlsanabria77/Desktop/17-02.2021 Instacat Basket Analysis 3/02 Data/Prepared Data/orders_products_combined.pkl')

In [4]:
# Turning folderpath into a string
path = r'/Users/jlsanabria77/Desktop/17-02.2021 Instacat Basket Analysis 3'

In [5]:
# 4. Check the shape of the imported data
df_ords_prods_combined.shape

(32434489, 10)

In [6]:
# 4.1 Check it is the same as the exported data in the exercise

-From the exercise notebook. It has the same size

df_merged_large['_merge'].value_counts()
Out[16]:
both          32434489
right_only           0
left_only            0
Name: _merge, dtype: int64

In [7]:
# 5. combine orders_products_combined with products dataset

In [6]:
# 5.2 Import products dataset
df_prods_checked = pd.read_csv(r'/Users/jlsanabria77/Desktop/17-02.2021 Instacat Basket Analysis 3/02 Data/Prepared Data/products_checked.csv', index_col = False)

In [9]:
# I will combine both datasets using a merge. Before I will check their shape to see how columns look

In [ ]:
df_prods_checked.head()

In [8]:
df_ords_prods_combined.head()

order_id  user_id eval_set  order_number  order_dow  order_hour_of_day  \
0   2539329        1    prior             1          2                  8   
1   2539329        1    prior             1          2                  8   
2   2539329        1    prior             1          2                  8   
3   2539329        1    prior             1          2                  8   
4   2539329        1    prior             1          2                  8   

   days_since_prior_order  product_id  add_to_cart_order  reordered  
0                     NaN         196                  1          0  
1                     NaN       14084                  2          0  
2                     NaN       12427                  3          0  
3                     NaN       26088                  4          0  
4                     NaN       26405                  5          0

In [ ]:
# 5.3 By looking at both df I can see only product_id column can be used as a link to create a merge.
df_orders_products_merged = df_ords_prods_combined.merge(df_prods_checked, on = 'product_id', indicator = True)

In [14]:
df_orders_products_merged.head()

order_id  user_id eval_set  order_number  order_dow  order_hour_of_day  \
0   2539329        1    prior             1          2                  8   
1   2398795        1    prior             2          3                  7   
2    473747        1    prior             3          3                 12   
3   2254736        1    prior             4          4                  7   
4    431534        1    prior             5          4                 15   

   days_since_prior_order  product_id  add_to_cart_order  reordered  \
0                     NaN         196                  1          0   
1                    15.0         196                  1          1   
2                    21.0         196                  1          1   
3                    29.0         196                  1          1   
4                    28.0         196                  1          1   

   Unnamed: 0 product_name  aisle_id  department_id  prices _merge  
0         195         Soda        77              7     9.0   both  
1         195         Soda        77              7     9.0   both  
2         195         Soda        77              7     9.0   both  
3         195         Soda        77              7     9.0   both  
4         195         Soda        77              7     9.0   both

In [17]:
# using the "merge flag" we can see that there is a full match of values
df_orders_products_merged['_merge'].value_counts()

both          32404859
right_only           0
left_only            0
Name: _merge, dtype: int64

In [ ]:
#  6. For the size of this merged dataframe I will choose pickle to export

df_orders_products_merged.to_pickle(os.path.join(path, '02 Data','Prepared Data', 'orders_products_merged.pkl'))